client/human/chmain.cpp

```diff
@@ -187,22 +187,22 @@ int mainConfigOptionsSetup(const std::vector<std::string>& args) {
             GetOptionsDB().Set<bool>("show-fps",                true);
         }
 
-    } catch (const std::invalid_argument& e) {
+    /* Restore this code when C++11 is adopted and rethrow e will preserve the stack trace.
+       Otherwise these Exception non-handlers just hide the stack trace and make debugging harder.
+    }
+    catch (const std::invalid_argument& e) {
         std::cerr << "main() caught exception(std::invalid_argument): " << e.what() << std::endl;
         boost::this_thread::sleep_for(boost::chrono::seconds(3));
-        return 1;
+        rethrow e;
```

Comment:
I'm having a hard time finding any material about this `rethrow` keyword. Can you point out the specification paper?